### ISSUE TITLE
[FEATURE REQ] Allow Resource Graph requests without enumerating tenants

### ISSUE THREAD

**Issue by aelij** [Azure/azure-sdk-for-net#44361]
### Library name

Azure.ResourceManager.ResourceGraph

### Please describe the feature.

Similar to #41023, which was closed however I think this can be viewed from a different angle.

ARG doesn't actually use the tenant ID in its API. So how about creating an extension method directly on `ArmClient`. For example:

```cs
Task<Response<ResourceQueryResult>> GetResourcesAsync(ArmClient, ResourceQueryContent, CancellationToken);
```

Moreover, using `TenantResource` in this context is wrong, as the query will **always** run in the user's default tenant, regardless of which `TenantResource` was used.

**Comment by HarveyLink** [comment]
Hi @aelij ,Thank you for the suggestions, our SDK is auto generated from the service spec, as the ARM definition, one auth token will only combine with single tanent so that the resource query could only run on default tenant.
On the other hand, based on the SDK design, a operation must extends on one resource, at this case, a tenenat resource is required here.

**Comment by aelij** [comment]
Hi @HarveyLink, sorry but I don't understand your explanation.

> the resource query could only run on default tenant

That's my point. If you run it on a non-default tenant returned from `GetTenants`, you will still run it on the default tenant, which is very confusing and erronous.

> operation must extends on one resource

My suggestion is to change this design, to either support extension methods directly on `ArmClient` itself for tenant-level resources or alternatively, create a `DefaultTenant` property on `ArmClient`.

/unresolve

**Comment by aelij** [comment]
> On the other hand, we will consider to add a Tenant property for the ArmClient.

Can you please keep this item open to track this request?
